### DISCUSSION TITLE
Number Pad Input

### DISCUSSION THREAD

**Discussion by RichardUUU** [iantrich/restriction-card#2] [Ideas]
I would love to have this card popup a number pad to enter a PIN (either fixed length or user specified). That would be much quicker than the phone keyboard.

**Comment by iantrich** [comment]
Potential basis: https://gist.github.com/ErikHellman/9e17f2ea6a78669294ef2af4bc3f5878

**Reply by ildar170975** [reply]
implemented in https://github.com/iantrich/restriction-card/releases/tag/1.2.14

**Comment by Kalfany** [comment]
You also could use parts of the HA [alarm panel](https://www.home-assistant.io/lovelace/alarm-panel/) for this ([source code](https://github.com/home-assistant/home-assistant-polymer/blob/dev/src/panels/lovelace/cards/hui-alarm-panel-card.ts))

**Comment by RichardUUU** [comment]
I've come up with a solution using Node RED that I documented [here](https://community.home-assistant.io/t/protect-a-card-with-a-ui-keypad-using-custom-button-and-node-red/161649).

**Comment by ildar170975** [comment]
https://github.com/iantrich/restriction-card/releases/tag/1.2.14
